ui/components/app/snaps/snap-ui-renderer/utils.ts

```diff
@@ -0,0 +1,17 @@
+import { Component } from '@metamask/snaps-sdk';
+import { COMPONENT_MAPPING } from './components';
+
+export type MapToTemplateParams = {
+  element: Component;
+  elementKeyIndex: number;
+  form?: string;
+};
+
+export const mapToTemplate = (params: MapToTemplateParams) => {
+  const { type } = params.element;
+  params.elementKeyIndex += 1;
+  const indexKey = `snap_ui_element_${type}__${params.elementKeyIndex}`;
+  // @ts-expect-error This is a problem with the types generated in the snaps repo.
+  const mapped = COMPONENT_MAPPING[type](params as any);
```

Comment:
I would like to avoid having `@ts-expect-error` here if the TypeScript is not able to Infer the key type properly. It is a string anyway at the end.
So, adding helper/utility type to handle this would help removing `@ts-expect-error`.

This can be the helper type:
```typescript
export type ComponentType =
  | 'panel'
  | 'heading'
  | 'text'
  | 'divider'
  | 'spinner'
  | 'image'
  | 'copyable'
  | 'row'
  | 'address'
  | 'button'
  | 'form'
  | 'input';
```
And the problematic line can be fixed with:
```typescript
const mapped = COMPONENT_MAPPING[type as ComponentType](params as any);
```
`ComponentType` can probably have a better name, anyway.

Reply:
I would rather not introduce this type even temporarily because we are trying not to duplicate code into the client repos. These types worked before the extension bumped the TypeScript version. 

The type export problems should be fixed when we ship https://github.com/MetaMask/snaps/pull/2120 AFAIK, so it seems fine to wait for that IMO.

Reply:
Alright. But, I would suggest that in the future we maybe wait for such stuff to get resolved/fixed/merged, then do a release. It feels weird to merge stuff with TypeScript errors expected and error ignore patterns that are in functional code, I'm fine having them only in the test files.
Anyway, I understand the issues here, so I'm not blocking PR, I can approve without fix for this.

Reply:
Agreed, but this problem started occurring while this PR was in development and not before (due to TypeScript being updated in the extension I believe). If it had been a problem from the start I would be fully in agreement.

Reply:
Sure, no problem. Let's fix it in the next release.